vault/rekey.go

```diff
@@ -629,24 +720,217 @@ func (c *Core) RecoveryRekeyUpdate(ctx context.Context, key []byte, nonce string
 		Value: []byte(c.recoveryRekeyConfig.Nonce),
 	}); err != nil {
 		c.logger.Error("error saving keyring canary", "error", err)
-		return nil, errwrap.Wrapf("failed to save keyring canary: {{err}}", err)
+		return logical.CodedError(http.StatusInternalServerError, errwrap.Wrapf("failed to save keyring canary: {{err}}", err).Error())
+	}
+
+	return nil
+}
+
+func (c *Core) RekeyVerify(ctx context.Context, key []byte, nonce string, recovery bool) (*RekeyVerifyResult, logical.HTTPCodedError) {
+	if recovery {
+		return c.RecoveryRekeyVerify(ctx, key, nonce)
+	}
+	return c.BarrierRekeyVerify(ctx, key, nonce)
+}
+
+func (c *Core) BarrierRekeyVerify(ctx context.Context, key []byte, nonce string) (ret *RekeyVerifyResult, retErr logical.HTTPCodedError) {
```

Comment:
Could this and RecoveryRekeyVerify be merged into the same function with the seal info and keys to verify passed into the function?

Reply:
Probably, will look.

Reply:
Updated.